presto-hive/src/main/java/com/facebook/presto/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -0,0 +1,703 @@
+            }
+        }
+
+        Table newTable = Table.builder(oldTable)
+                .setDataColumns(newDataColumns.build())
+                .build();
+        replaceTable(databaseName, tableName, newTable, null);
+    }
+
+    @Override
+    public void dropColumn(String databaseName, String tableName, String columnName)
+    {
+        verifyCanDropColumn(this, databaseName, tableName, columnName);
+        Table oldTable = getTableOrElseThrow(databaseName, tableName);
+
+        if (!oldTable.getColumn(columnName).isPresent()) {
+            SchemaTableName name = new SchemaTableName(databaseName, tableName);
+            throw new ColumnNotFoundException(name, columnName);
+        }
+
+        ImmutableList.Builder<Column> newDataColumns = ImmutableList.builder();
+        for (Column fieldSchema : oldTable.getDataColumns()) {
+            if (!fieldSchema.getName().equals(columnName)) {
+                newDataColumns.add(fieldSchema);
+            }
+        }
+
+        Table newTable = Table.builder(oldTable)
+                .setDataColumns(newDataColumns.build())
+                .build();
+        replaceTable(databaseName, tableName, newTable, null);
+    }
+
+    @Override
+    public Optional<Partition> getPartition(String databaseName, String tableName, List<String> partitionValues)
+    {
+        try {
+            GetPartitionResult result = glueClient.getPartition(new GetPartitionRequest()
+                    .withDatabaseName(databaseName)
+                    .withTableName(tableName)
+                    .withPartitionValues(partitionValues));
+            return Optional.of(GlueToPrestoConverter.convertPartition(result.getPartition()));
+        }
+        catch (EntityNotFoundException e) {
+            return Optional.empty();
+        }
+        catch (AmazonServiceException e) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+        }
+    }
+
+    @Override
+    public Optional<List<String>> getPartitionNames(String databaseName, String tableName)
+    {
+        Table table = getTableOrElseThrow(databaseName, tableName);
+        List<Partition> partitions = getPartitions(databaseName, tableName, WILDCARD_EXPRESSION);
+        return Optional.of(buildPartitionNames(table.getPartitionColumns(), partitions));
+    }
+
+    /**
+     * @param parts List of partition values to filter on which can be full or partial.
+     * ex: partition keys = ['a', 'b', 'c']
+     * parts could be:
+     * ['1','2','3'] or
+     * ['', '2', '']
+     * @return a list of partition names.
+     */
+    @Override
+    public Optional<List<String>> getPartitionNamesByParts(String databaseName, String tableName, List<String> parts)
+    {
+        Table table = getTableOrElseThrow(databaseName, tableName);
+        String expression = GlueExpressionUtil.buildExpression(table.getPartitionColumns(), parts);
+        List<Partition> partitions = getPartitions(databaseName, tableName, expression);
```

Comment:
`getPartitionNamesByParts()` is supposed to be a fast to list just the partition names as it is used during query planning. We then fetch the full partition metadata in a pipelined manner as the query executes. Fetching partition metadata can be slow and expensive (especially with the traditional Hive metastore). Even if the metastore is very fast, tables might have millions of partitions. (Though looks like Glue limits to 20k partitions per table. That would be nice.)

But it appears Glue doesn't have an API to just fetch the names, so this might be the best we can do. One downside is that we will fetch the partition metadata twice, since there's no caching. (`HiveMetadataFactory` uses `CachingHiveMetastore.memoizeMetastore()` when it creates `SemiTransactionalHiveMetastore`, but that won't help since the full partition objects are discarded by this method.)

Reply:
Yea unfortunately there is no API for getting just names. There could be a couple ways to optimize this such as messing with the partition caching in ```CachingHiveMetastore``` but that would probably conflict with other metastores. Probably save this issue for another time...